### DISCUSSION TITLE
Problem using vector graphics to maintain image quality

### DISCUSSION THREAD

**Discussion by amirian** [ImageMagick/ImageMagick#2050] [Help]
Hi,
In order to use some patterns in different scales without loosing quality, I converted the source file to a BW svg format and used IM to generate the image in my own scale, but the output quality isn't what I expected! The svg file can be zoomed in any viewer to any degree with no damage in smoothness. I changed the range of colors to 4 but the problem still remains:
magick pattern.svg -sample 6000 -depth 2 -colors 4 -define png:bit-depth=2 PNG8:output.png
I attach one of the patterns to test it yourself.
[pattern.zip](https://github.com/ImageMagick/ImageMagick/files/4661843/pattern.zip)

**Comment by snibgo** [comment]
Use "-density 300" or some other (larger) number before reading the svg.

**Reply by amirian** [reply]
Thank you. I'm getting somehow confused with the density option: 300 is good with "-sample 6000", but my actual sample is now 1260. when I use -density 300 or higher (even up to 1000) with -sample 1260, the smoothness is reduced. Comparing the 1260 pxl output in actual size (right) with the big 6000 pxl with relatively same scaled in viewer(left), the actual size view of 1260 is not acceptable.
![screen](https://user-images.githubusercontent.com/1963455/82556134-66235e80-9b7e-11ea-919b-b2b29f534f3b.png)

**Comment by snibgo** [comment]
Why are you sampling, after rasterizing the SVG? Best quality usually comes from rasterizing the SVG at whatever density gives you the size you want, so there is no need for sampling (or resizing) after rasterizing.

**Reply by amirian** [reply]
So, you mean the best solution for a final raster image of 1260x2036, I should set the density to 87, to get a 1265x2026 image, and then cut that with -extent 1260x2038. Correct? But this is also not so acceptable in 2 colors (but in 4 colors) with the following command:
magick -density 87 pattern.svg -extent 1260x2038 -depth 1 -colors 2 -define png:bit-depth=1 PNG8:output1.png
![output1](https://user-images.githubusercontent.com/1963455/82560501-d209c500-9b86-11ea-94d1-e80f0adbd772.png)

**Reply by amirian** [reply]
I installed inkscape on Windows and put in the PATH, but just not sure if IM uses it, so I switched to linux. The result of rasterizing svg with IM through inkscape was TERRIBLE! However, I'm satisfied with the last result with 4 colors, thank you.

**Comment by snibgo** [comment]
If you want 1260x2036 then I suggest a "-density" of 87.414. This gives a 1271x2036 image. You can "-crop" this to 1260x2036.

**Reply by amirian** [reply]
Appreciated and accepted. My last question was: That's good for 4 colors. The problem is now with 2 colors: with density 87*4 and zooming the viewer to 25% (right), I see a better quality than with density 87 and zooming to actual size(left). Perhaps that's for the capability of the viewer, but I expect to get the best possible solution with density 87 and zooming to actual size.
![screen](https://user-images.githubusercontent.com/1963455/82563237-ec926d00-9b8b-11ea-9c33-1b21a101874c.png)

**Comment by snibgo** [comment]
What software rasterizes your SVG? "-verbose" may tell you this. I suggest always using Inkscape to rasterize SVG, unless there is a good reason (eg performance) for using MSVG or RSVG.

To use Inkscape, simply install it and ensure it is on your system path. IM will find it, and use it.
